lua/barbar/render.lua

```diff
@@ -39,170 +40,13 @@ local utils = require'barbar.utils'
---- @return string
---- @diagnostic disable-next-line:unused-function,unused-local
-local function groups_to_raw_string(groups)
-  local result = ''
-
-  for _, group in ipairs(groups) do
-    result = result .. group.text
-  end
-
-  return result
-end
-
---- Insert `others` into `groups` at the `position`.
---- @param groups barbar.render.group[]
---- @param position integer
---- @param others barbar.render.group[]
---- @return barbar.render.group[] with_insertions
-local function groups_insert(groups, position, others)
-  local current_position = 0
-
-  local new_groups = {}
-
-  local i = 1
-  while i <= #groups do
-    local group = groups[i]
-    local group_width = strwidth(group.text)
-
-    -- While we haven't found the position...
-    if current_position + group_width <= position then
-      table_insert(new_groups, group)
-      i = i + 1
-      current_position = current_position + group_width
-
-    -- When we found the position...
-    else
-      local available_width = position - current_position
-
-      -- Slice current group if it `position` is inside it
-      if available_width > 0 then
-        table_insert(new_groups, {
-          text = strcharpart(group.text, 0, available_width),
-          hl = group.hl,
-        })
-      end
-
-      -- Add new other groups
-      local others_width = 0
-      for _, other in ipairs(others) do
-        local other_width = strwidth(other.text)
-        others_width = others_width + other_width
-        table_insert(new_groups, other)
-      end
-
-      local end_position = position + others_width
-
-      -- Then, resume adding previous groups
-      -- table.insert(new_groups, 'then')
-      while i <= #groups do
-        local previous_group = groups[i]
-        local previous_group_width = strwidth(previous_group.text)
-        local previous_group_start_position = current_position
-        local previous_group_end_position   = current_position + previous_group_width
-
-        if previous_group_end_position <= end_position and previous_group_width ~= 0 then
-          -- continue
-        elseif previous_group_start_position >= end_position then
-          -- table.insert(new_groups, 'direct')
-          table_insert(new_groups, previous_group)
-        else
-          local remaining_width = previous_group_end_position - end_position
-          local start = previous_group_width - remaining_width
-          local end_  = previous_group_width
-          table_insert(new_groups, { hl = previous_group.hl, text = strcharpart(previous_group.text, start, end_) })
-        end
-
-        i = i + 1
-        current_position = current_position + previous_group_width
-      end
-
-      break
-    end
-  end
-
-  return new_groups
-end
-
 --- Create valid `&tabline` syntax which highlights the next item in the tabline with the highlight `group` specified.
 --- @param group string
 --- @return string syntax
-local function hl_tabline(group)
+local function wrap_hl(group)
```

Comment:
We already use the `hl_` prefix for actual hl strings, better to differentiate the function that creates them.

Reply:
Good thought.